File: VS/lab3/WarehouseEntry.cpp
#include "WarehouseEntry.hpp"
#include "Warehouse.hpp"

template<typename T>
WarehouseEntry<T>::WarehouseEntry(int id, std::string name)
    : m_id(id), m_name(name) {
}

template<typename T>
const std::string& WarehouseEntry<T>::GetName() const {
    return m_name;
}

template<typename T>
void WarehouseEntry<T>::SetName(std::string name) {
    m_name = name;
}

template<typename T>
int WarehouseEntry<T>::GetId() const {
    return m_id;
}

template<typename T>
void WarehouseEntry<T>::SetId(int id) {
    m_id = id;
}

template<typename T>
void WarehouseEntry<T>::AssignWarehouse(const Warehouse* warehouse) {
    if (warehouse) {
        m_assignedWarehouseId = warehouse->GetId();
        m_assignedWarehouse = warehouse;
    }
}

File: lab7/Java/src/warehouse/WarehouseEntry.java
package warehouse;

import request.SupplyRequestDetail;

public abstract class WarehouseEntry<T extends Number> {

    String name;
    protected int id;
    protected int assignedWarehouseId;
    protected Warehouse assignedWarehouse;

    public WarehouseEntry(int id, String name){
        this.id = id;
        this.name = name;
    }

    public int getId(){
        return id;
    }
    public String getName(){
        return name;
    }
    
    public void assignWarehouse(Warehouse warehouse){
        if(warehouse!=null){
            this.assignedWarehouseId = warehouse.getId();
            this.assignedWarehouse = warehouse;
        }
    }
    public abstract Boolean isMatches(SupplyRequestDetail detail);
    public abstract Boolean isEmpty();
    public abstract void increase(Number v);
    public abstract T decrease(Number v);
}
